File: 프로그래머스/unrated/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length];//answer배열을 생성하고 길이는 arr배열의 길이와 동일하게 설정
        
        for(int i=0; i<arr.length; i++){
           if(arr[i] < 50 && arr[i] % 2 != 0){ //50보다 작은 홀수이면
               answer[i] = arr[i] * 2; //answer에 저장될 값을 2로 곱합니다.
           }else if(arr[i] >= 50 && arr[i] % 2 ==0){//50보다 크거나 같은 짝수일때
               answer[i] = arr[i] / 2; //값을 2로 나누어 answer 배열에 저장한다.
           }else{
               answer[i] = arr[i];
           }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.js
//map 사용 - 다른사람 풀이 참조
//function solution(arr) {
//    return arr.map(a => {
//        if(a >= 50 && a %2 ===0) return Math.floor(a/2)
//        if(a < 50 && a %2 === 1) return a*2
//        return a
//    })
//}

function solution(arr){
    var answer = [];
    
    for(var i=0; i<arr.length; i++){
        if(arr[i] < 50 && arr[i] % 2 != 0){
            answer.push(arr[i]*2)
        }else if(arr[i] >= 50 && arr[i] % 2 == 0){
            answer.push(arr[i]/2)
        }else{
            answer.push(arr[i])
        }
    }
    return answer;
}